README.md

```diff
@@ -2,64 +2,76 @@
 
 Serve DNS authoritative responses... with Erlang.
 
+[![Erlang/OTP Versions](https://img.shields.io/badge/erlang%2Fotp-27%7C28-blue)](https://www.erlang.org)
 [![Build Status](https://github.com/dnsimple/erldns/actions/workflows/ci.yml/badge.svg)](https://github.com/dnsimple/erldns/actions/workflows/ci.yml)
 [![Module Version](https://img.shields.io/hexpm/v/erldns.svg)](https://hex.pm/packages/erldns)
+[![Hex Docs](https://img.shields.io/badge/hex-docs-lightgreen.svg)](https://hexdocs.pm/erldns/)
+[![Hex Downloads](https://img.shields.io/hexpm/dt/erldns.svg)](https://hex.pm/packages/erldns)
+[![Coverage Status](https://coveralls.io/repos/github/dnsimple/erldns/badge.svg?branch=main)](https://coveralls.io/github/dnsimple/erldns?branch=main)
 
-## Requirements
+This application consists of three main subsystems:
 
-Erlang/OTP 27+
+- `m:erldns_zones`
+The system responsible for loading and caching zone data. Zones are loaded from JSON, example files are in the `priv/` directory. You can also write new systems to load zones by writing the zones directly to the zone cache using `erldns_zone_cache:put_zone/1`.
+
+- `m:erldns_pipeline`
+The system responsible for processing incoming DNS queries. It declares a pipeline of sequential transformations to apply to the incoming query until a response is constructed.
+
+- `m:erldns_listeners`
+The system responsible for listening for incoming DNS queries. The system is designed to be able to listen on multiple ports and interfaces and supports both UDP and TCP, Unix network stack optimisations, and high parallelism.
+
+There is also an administrative API for querying the current zone cache and for basic control.
+You can read more about it at `m:erldns_admin`.
+
+## Instrumentation
+
+[Telemetry](https://hex.pm/packages/telemetry) is used to instrument the code.
+
+## Configuration
+
+An example configuration file can be found in `erldns.example.config`.
 
 ## Building
 
 To build:
 
-```shell
+```sh
 make
 ```
 
 To start fresh:
 
-```shell
+```sh
 make fresh
 make
 ```
 
-## Zones
```

Comment:
I see the current content being moved to the "subsystem", and this is fine. But even if that's moved, we should keep a more informative Zones section with proper documentation targeting users.

The subsystem is highly technical and implementer oriented. Imagine I will download this server from Docker or I just want to run it. It will be more friendly to have a section that explains how to add/use zones. And perhaps what format and what expectation. It may delegate to the `/priv/zonefile.md` (renamed README) if you prefer. Or you move that part here.